docs/containers-registries.conf.5.md

```diff
@@ -18,42 +18,49 @@ VERSION 2 is the latest format of the `registries.conf` and is currently in
-point to `example.com/image`.
-
-By default container runtimes use TLS when retrieving images from a registry.
-If the registry is not setup with TLS, then the container runtime will fail to
-pull images from the registry. If you set `insecure = true` for a registry or a
-mirror you overwrite the `insecure` flag for that specific entry.  This means
-that the container runtime will attempt use unencrypted HTTP to pull the image.
-It also allows you to pull from a registry with self-signed certificates.
-
-If you set the `unqualified-search = true` for the registry, then it is possible
-to omit the registry hostname when pulling images.  This feature does not work
-together with a specified `prefix`.
+The configuration is a sequence of `[[registry]]` entries.  Given an input image
+name, a single entry is chosen; that entry must have a `prefix` field set to be
+a prefix of the input image name (taking into account namespace/repo/tag/digest
+separators), and (only) the entry with the longest matching `prefix` field
+is used.
+
+As a special case, the `prefix` field can be missing; if so, it defaults to the value
+of the `location` field.
+
+By default, container runtimes require TLS when retrieving images from a registry.
+The `insecure = true` option allows using unencrypted HTTP as well as TLS connections
+with untrusted certificates.
+
+The `location` field specifies the physical location of the registry.
+This is typically the same as `prefix` (in which case `prefix` can be omitted),
+but it can be different to redirect operations with the image to a different physical
+server (but the image will continue to use the original `prefix`-based name, not the
+result of the redirection).  If `prefix = "example.com/foo"`, `location = "example.net/bar"`
+and the `example.com/foo/myimage:latest` image is requested, the image will actually be
+pulled from `example.net/bar/myimage:latest`.
+
+In addition to the single `location`, a registry can have (partial) mirrors configured
+in a `mirror` array.  The mirrors will be evaluated in order during an image pull;
+the first one that can be contacted and contains the image will be used (and if none
+of the mirrors contains it, the "registry.location" field above is tried last).
+
+The mirror's `location` field has the same rewriting semantics as the registry's `location` field above.
+Furthermore, it is possible to specify the `insecure` flag per registry mirror, too.
+
+*Note*: Both the `location` field and mirrors are currently ignored when pushing to a registry;
+that may change in the future.
+
+If the `unqualified-search = true` option is set, the registry can be used when pulling
+an hostname-unqualified image. All registries with `unqualified-search` set to true are tried
+in the order they appear in the configuration file.
 
 If `blocked = true` then it is not allowed to pull images from that registry.
 
 ### EXAMPLE
 
 ```
 [[registry]]
-location = "example.com"
+location = "example.com/foo"
```

Comment:
I prefer to give examples in the descriptions of the each registry attribute. This specific example will teach users how the syntax looks like but not yet the semantics.

Reply:
The `location` field documentation now has a clearly-identifiable example (using the real-world formatting instead of in-line text inside a paragraph), and the example is more practical (actually saying `internal-registry-for-example` instead of an unmotivated mapping from `example.com` to `example.net`).